gcloud/_helpers.py

```diff
@@ -159,6 +165,41 @@ def _app_engine_id():
     return app_identity.get_application_id()
 
 
+def _file_project_id():
+    """Gets the project id from the credentials file if one is available.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from JSON credentials file if value exists,
+              else ``None``.
+    """
+    credentials_file_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS')
+    if credentials_file_path:
+        with open(credentials_file_path, 'r') as credentials_file:
```

Comment:
JSON is required to be encoded as UTF-8, which makes binary mode seem best.  Does the Python3 `json` module expect a binary stream?